diff --git a/ggc-app/src/ggc/app/main/DoAdvanceDate.java b/ggc-app/src/ggc/app/main/DoAdvanceDate.java
index 9cd18e0..c2a5c76 100644
--- a/ggc-app/src/ggc/app/main/DoAdvanceDate.java
+++ b/ggc-app/src/ggc/app/main/DoAdvanceDate.java
@@ -2,8 +2,14 @@ package ggc.app.main;
 
 import pt.tecnico.uilib.menus.Command;
 import pt.tecnico.uilib.menus.CommandException;
+
+import java.io.StringReader;
+import java.util.Scanner;
+
 import ggc.WarehouseManager;
-//FIXME import classes
+
+import ggc.app.exceptions.InvalidDateException;
+import ggc.exceptions.NoSuchDateException;
 
 /**
  * Advance current date.
@@ -12,12 +18,20 @@ class DoAdvanceDate extends Command<WarehouseManager> {
 
   DoAdvanceDate(WarehouseManager receiver) {
     super(Label.ADVANCE_DATE, receiver);
-    //FIXME add command fields
+    addIntegerField("date", Prompt.daysToAdvance());
   }
 
   @Override
   public final void execute() throws CommandException {
-    //FIXME implement command
+    try{
+      _receiver.updateDate(integerField("date"));
+      // _display.popup(Message.currentDate(date));
+    }
+
+    catch (NoSuchDateException e) {
+      throw new InvalidDateException(e.getDate());
+    }
+
   }
 
 }
diff --git a/ggc-app/src/ggc/app/main/DoDisplayDate.java b/ggc-app/src/ggc/app/main/DoDisplayDate.java
index b36af32..4a68cdb 100644
--- a/ggc-app/src/ggc/app/main/DoDisplayDate.java
+++ b/ggc-app/src/ggc/app/main/DoDisplayDate.java
@@ -2,6 +2,9 @@ package ggc.app.main;
 
 import pt.tecnico.uilib.menus.Command;
 import pt.tecnico.uilib.menus.CommandException;
+
+import java.io.BufferedWriter;
+
 import ggc.WarehouseManager;
 
 /**
@@ -14,8 +17,9 @@ class DoDisplayDate extends Command<WarehouseManager> {
   }
 
   @Override
-  public final void execute() throws CommandException {
-    //FIXME implement command
+  public final void execute() {
+    int date = _receiver.getDate();
+    _display.popup(Message.currentDate(date));
   }
 
 }
diff --git a/ggc-core/src/ggc/Warehouse.java b/ggc-core/src/ggc/Warehouse.java
index 86c94d2..b82cb02 100644
--- a/ggc-core/src/ggc/Warehouse.java
+++ b/ggc-core/src/ggc/Warehouse.java
@@ -13,6 +13,8 @@ public class Warehouse implements Serializable {
   /** Serial number for serialization. */
   private static final long serialVersionUID = 202109192006L;
 
+  private int _date;
+
   // FIXME define attributes
   // FIXME define contructor(s)
   // FIXME define methods
@@ -22,6 +24,18 @@ public class Warehouse implements Serializable {
    * @throws IOException
    * @throws BadEntryException
    */
+
+  public int getDate() {
+    return _date;
+  }
+
+  public void updateDate(int date) throws NoSuchDateException {
+    if (date <= 0) {
+      throw new NoSuchDateException(date);
+    }
+    _date += date;
+  }
+   
   void importFile(String txtfile) throws IOException, BadEntryException /* FIXME maybe other exceptions */ {
     //FIXME implement method
   }
diff --git a/ggc-core/src/ggc/WarehouseManager.java b/ggc-core/src/ggc/WarehouseManager.java
index f98f039..0e7e67d 100644
--- a/ggc-core/src/ggc/WarehouseManager.java
+++ b/ggc-core/src/ggc/WarehouseManager.java
@@ -18,6 +18,14 @@ public class WarehouseManager {
   //FIXME define constructor(s)
   //FIXME define other methods
 
+  public int getDate() {
+    return _warehouse.getDate();
+  }
+
+  public void updateDate(int date) throws NoSuchDateException {
+    _warehouse.updateDate(date);
+  }
+
   /**
    * @@throws IOException
    * @@throws FileNotFoundException
diff --git a/ggc-core/src/ggc/exceptions/NoSuchDateException.java b/ggc-core/src/ggc/exceptions/NoSuchDateException.java
new file mode 100644
index 0000000..f5452ca
--- /dev/null
+++ b/ggc-core/src/ggc/exceptions/NoSuchDateException.java
@@ -0,0 +1,13 @@
+package ggc.exceptions;
+
+public class NoSuchDateException extends Exception{
+    private final int _date;
+
+    public NoSuchDateException(int date) {
+        _date = date;
+    }
+
+    public int getDate() {
+        return _date;
+    }
+}
